### PR TITLE
Rfc ecs milestone1

### PR DESCRIPTION
**Pull request by mcasperson**
# About

This is a post about the proposed ECS integration being delivered by the steps team.

# Pre-requisites

- [x] The draft is complete and this post is ready to be reviewed.
- [x] I'm familiar with the [writing for Octopus TL;DR](https://style.octopus.com/writing-at-octopus-tldr).
- [x] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [x] The images I've included follow the [rules for working with images](https://style.octopus.com/images#rules-for-screenshots).

## How to review this PR

This post represents a new style of content from the steps team. It is intended to:

* Generate excitement and feedback from customers
* Inform internal teams of our plans and reasons
* Provide context for an associated pitch
* But not promise to deliver any new functionality

There are some competing concerns in the list above, which should be taken into account when reviewing the content.

## Publication date

ASAP.

### CODE REVIEW

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+We want Octopus to be your first choice regardless of whether you deploy on-premises, to the cloud, or to a hybrid of both. This means providing first class support for more PaaS offerings to make your complex deployments easy.
+
+To reach this goal, Octopus has established a new internal team dedicated to integrating Octopus with popular PaaS offerings. After spending some months focused on developing a new framework to deliver our PaaS integrations, we are now in a position to share our goals and plans with our customers, partners, and other internal Octopus departments. This blog is what we hope will be the first of many Request For Comment (RFC) posts where we discuss proposed new functionality and provide an opportunity for those interested to offer feedback.
+
+One PaaS service that we have been repeatedly asked to support is AWS ECS, and this post will discuss the proposed new steps and targets that we are currently discussing.
```
Comment (egorpavlikhin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:19]: Discuss/discussing repetition reads a bit wrong here.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+## Conclusion
+
+In summary, the first milestone of our proposed ECS support includes:
+
+* A new target to model environments and tenanted deployments.
+* A new step combining the creation/updating of a task definition and creation/updating of a service.
+* The ability to convert the opinionated step into a raw CloudFormation template.
+* A simplified deployment workflow across environments supporting environment specific variables, channels, and lifecycles.
+
+Thankyou for reading this post. We hope you are as excited about the proposed new ECS functionality as we are, and any feedback you may have is greatly appreciated.
```
Comment (egorpavlikhin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:155]: > Thankyou

Typo.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+author: matthew.casperson@octopus.com
+visibility: private
+published: 2999-01-01
+metaImage: 
+bannerImage: 
+tags:
+ - Octopus
+---
+
+Octopus is (among other things) a deployment tool. We have had industry leading functionality for deploying to your on-premises infrastructure from the beginning, and we've powered Azure application deployments for years. Our Kubernetes steps expanded our deployment capabilities to all major cloud providers, but since those steps were released in 2018 Octopus hasn't enabled first class deployments to any new cloud Platform as a Service (PaaS) offerings.
```
Comment (andrewabest) [inline blog/2021-06/rfc-ecs-milestone1/index.md:13]: I've had a go at editing this introduction, with a couple of goals:
1. Don't admit we haven't shipped any new steps since 2018, I'm not sure if that is valuable here
2. Use "cloud native services" instead of "PaaS" - as many techos view "PaaS" as a subset of cloud services, and distinguish it from FaaS, CaaS and other "aaS" things.

```
Octopus is a tool that enables world-class deployments. We have had industry leading functionality for deploying to your on-premises infrastructure from the beginning, and we've powered Azure application deployments for years. More recently we expanded into other major cloud service providers, allowing you to use tools like Kubernetes and Terraform to deploy to Azure and AWS. But we can do more - there are still a number of cloud native services that Octopus does not yet provide first class integrations for. 

We want Octopus to be your first choice regardless of whether you deploy on-premises, to the cloud, or to a hybrid of both. This means providing first class support for more cloud native service offerings to make your complex deployments easy.

To reach this goal, Octopus has established a new internal team dedicated to integrating Octopus with the most popular cloud native services. After spending some months focused on developing a new framework to enable the rapid delivery of these integrations, we are now in a position to share our goals and plans with our customers, partners, and other internal Octopus departments. This blog is what we hope will be the first of many Request For Comment (RFC) posts where we discuss proposed new functionality and provide an opportunity for those interested to offer feedback.

One cloud service that we have been repeatedly asked to support is AWS ECS, and this post will outline some new steps and targets that we are currently discussing.
```

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+*Step mockup showing Docker image selection and task definition inputs.*
+
+The same step defines the values contributed to the properties of a service:
+
+![](https://via.placeholder.com/500x300 "width=500")
+
+*Step mockup showing service properties.*
+
+An ECS deployment will then execute the following process:
```
Comment (andrewabest) [inline blog/2021-06/rfc-ecs-milestone1/index.md:61]: I'm not sure if this aligns with what I was expecting. I had thought we were going to create a new task definition per release and treat them as immutable - doing this would feel "safer" (less chance of bugs in the diffing code), and we could tag the definition with the Octopus release if we wanted traceability from the service back to Octopus. From a user perspective it also feels very straightforward - each new release is a new task definition.

Can you think of a reason why diffing would provide a better experience then treating them as immutable?

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+Behind the scenes the new step will generate CloudFormation templates, which will then be executed for you. This ensures all resources can be [audited](https://octopus.com/blog/ten-pillars-of-pragmatic-deployments#auditable-deployments) and [tracked](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-stack-drift.html) with existing CloudFormation tooling.
+
+However, any opinionated step will eventually meet a use case that it simply does not support. For those with specialized requirements, or who find they outgrow the proposed step, we'll offer the ability to convert the opinionated step into a raw CloudFormation template. 
+
+Simply select the convert option in the overflow menu, and the step will be converted into a CloudFormation deployment step, giving you complete control over your ECS deployments, without having to recreate them from scratch:
+
+![](convert.png "width=500")
+
+## What is the scope of the first ECS milestone?
```
Comment (andrewabest) [inline blog/2021-06/rfc-ecs-milestone1/index.md:122]: It feels like this part should be way higher up - this is some of the key stuff we want feedback on, and it may get a bit "lost" all the way down here. I wonder if this can fit above the "A new ECS Target" section?

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 4 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:61]: Thinking through the process, it was clear that a task definition could contain environment specific details, especially environment variables. There isn't a way to map a task definition to a release unless we limit the use of variables.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+* Limiting the step to deploying to Fargate only.
+* Only support rolling deployments, and not support blue/green deployments.
+* Not provide the ability to build a new load balancer.
+* Exclude auto-scaling settings.
+* Exclude app mesh and firelens settings.
+* Exclude service auto-discovery settings.
+* Only create a service, and not support tasks or scheduled tasks.
+
+We do however see these features being included in subsequent milestones, so watch for a new RFC post covering these.
```
Comment (andrewabest) [inline blog/2021-06/rfc-ecs-milestone1/index.md:134]:
```suggestion
These features are likely to be included in subsequent milestones, so watch for new RFC posts covering these.
```

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 4 above)
Comment (andrewabest) [reply blog/2021-06/rfc-ecs-milestone1/index.md:61]: Ah of course - the task definition has to change between environments. I understand :)

Comment (andrewabest) [reply blog/2021-06/rfc-ecs-milestone1/index.md:61]: What if we created a new one per deployment?

Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:61]: 👍 That might be easier to explain, and would translate more directly to the underlying cloudformation templates. I'll update the process to reflect that.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
+---
+title: RFC - ECS integration with Octopus
+description: We're proposing new integrations with ECS in Octopus, and want your feedback!
+author: matthew.casperson@octopus.com
+visibility: private
+published: 2999-01-01
+metaImage: 
+bannerImage: 
+tags:
+ - Octopus
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:10]: This has been changed.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+---
+
+Octopus is (among other things) a deployment tool. We have had industry leading functionality for deploying to your on-premises infrastructure from the beginning, and we've powered Azure application deployments for years. Our Kubernetes steps expanded our deployment capabilities to all major cloud providers, but since those steps were released in 2018 Octopus hasn't enabled first class deployments to any new cloud Platform as a Service (PaaS) offerings.
+
+We want Octopus to be your first choice regardless of whether you deploy on-premises, to the cloud, or to a hybrid of both. This means providing first class support for more PaaS offerings to make your complex deployments easy.
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:15]: @robpearson - Is this something that affects the marketing team?

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 1 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:19]: Some of the external feedback was to make sure this post was not mistaken for promoising any new functionality, hence the emphasis on "discussing" and "proposing" a solution. But exploring would be a good choice here.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+![](lifecycles.png "width=500")
+
+
+### Fall back to CloudFormation
+
+Behind the scenes the new step will generate CloudFormation templates, which will then be executed for you. This ensures all resources can be [audited](https://octopus.com/blog/ten-pillars-of-pragmatic-deployments#auditable-deployments) and [tracked](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-stack-drift.html) with existing CloudFormation tooling.
+
+However, any opinionated step will eventually meet a use case that it simply does not support. For those with specialized requirements, or who find they outgrow the proposed step, we'll offer the ability to convert the opinionated step into a raw CloudFormation template. 
+
+Simply select the convert option in the overflow menu, and the step will be converted into a CloudFormation deployment step, giving you complete control over your ECS deployments, without having to recreate them from scratch:
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:118]: If we update the cloudformation step to include additional package references (much like the Kubernetes YAML deployment step), I think we can provide a one-to-one conversion here. Each image in the task definition will become an additional package reference, and the CloudFormation template can contain both the task definition and service details, with appropiate dependencies.

An example of a single template including both the task definition and service can be found at https://octopus.com/blog/ecs-canary-deployments#the-complete-template.

[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:118]: I'll add a note that the CloudFormation step will be updated as part of this first milestone.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 3 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:13]: The value of mentioning the gap between new steps was more for internal readers, as I'm still not sure most people realise just how long it has been since we have added support for a new service. But I'm happy to not call it out specifically.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+## When will this be released?
+
+We're still in the early planning stages, and ECS support is not a confirmed feature at this point, so we can not provide a release date. Keep an eye on the [blog](https://octopus.com/blog/) for further announcements.
+
+## We want your feedback!
+
+ECS support is still in the planning phases, so now is a great time to shape the future of this new feature with your feedback. We have created a GitHub issue here *TODO* to capture the discussion.
+
+Specifically we want to know if the proposed step and target will work for your ECS deployments, as well as learning what your ECS architecture looks like. Do you have multiple clusters? Do you have multiple AWS accounts? What kinds of applications are you deploying? What ECS deployment challenges do you wish Octopus could solve for you? This is all great feedback that will help us deliver the best solution we can.
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:143]: I'm happy to leave style changes like this to the editing pass.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+![](https://via.placeholder.com/500x300 "width=500")
+
+*An ECS target mockup.*
+
+### A new service deployment step
+
+At a high level, there are three components required to deploy an application to an ECS cluster.
+
+First you need a Docker image. We envision your Continuous Integration (CI) server will continue to build and deploy these images to a Docker registry.
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:41]: I'll add a note that the CI server should be tagging the images.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 5 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:121]: I deliberatly left the link to the GitHub issue down the bottom to encourage feedback from those who finished the post, and it seemed appropriate to put this list of limitations near that link and the list of things we were looking for feedback on.

I'd also like to draw readers in with what we are going to do and the cool processes they will unlock before we give them all the things we are going to skip.

Comment (andrewabest) [reply blog/2021-06/rfc-ecs-milestone1/index.md:121]: Fair enough :)

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+
+At a high level, there are three components required to deploy an application to an ECS cluster.
+
+First you need a Docker image. We envision your Continuous Integration (CI) server will continue to build, tag, and deploy these images to a Docker registry.
+
+A [task definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html) then references a specific image tag and defines many of the settings for the resulting container such as memory and CPU requirements, environment variables, exposed ports and more. Task definitions are immutable, and each new image tag must be captured by a new version of a task definition.
+
+A [service](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service_definition_parameters.html) then references a task definition, along with additional runtime details such as how many instances to run, how the instances are distributed through the cluster, which VPC to run in, load balancers and scaling requirements.
+
+Our proposed new step provides an opinionated deployment workflow that combines a Fargate task definition and service into a single step.
```
Comment (adrianhowchin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:47]: I was a little confused to see Fargate mentioned for the first time here - is this meant to be the case?

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+[Amazon describes Elastic Container Service (ECS) as](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html):
+
+> a highly scalable, fast container management service that makes it easy to run, stop, and manage containers on a cluster. Your containers are defined in a task definition that you use to run individual tasks or tasks within a service.
+
+Octopus already has many features in place to coordinate container deployments, including the ability to consume Docker images via feeds (including ECR feeds) and securely storing AWS credentials via an account. By scripting against the AWS CLI it has been *possible*, if not convenient, to deploy to ECS today. But we can do better.
+
+### A new ECS target
+
+The proposed ECS support starts with a new target that represents an ECS cluster. This target references the AWS credentials used to access an ECS cluster, the AWS region, and the name of the cluster:
```
Comment (adrianhowchin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:31]: If I understand this correctly, you're saying that an ECS cluster will be represented as one Deployment Target in Octopus. Given what we charge for a single Deployment Target, I don't feel this captures the value we would offer through this work. I'm not sure what the right level is though. 

If we differentiated Target types (normal Target, ECS Target) then we could charge differently for them, but... that sounds painful as customers would need to have some idea up-front about their patterns of usage.

Comment (adrianhowchin) [reply blog/2021-06/rfc-ecs-milestone1/index.md:31]: I really like the technical benefits of defining them as the Target (listed under "Why use Targets" below). I'm just not sure the best way forward from a business perspective.

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+
+![](targets.png "width=500")
+
+### Abstracting away task definition versions
+
+If you have ever had to deploy a new Docker image by first creating a new task definition version and then updating the service to reference it, you will appreciate how tedious manual ECS deployments can be.
+
+Our goal is for a new ECS deployment to involve nothing more than creating an Octopus release and selecting the new Docker image tags to include in it. By creating a new task definition with each deployment we remove the need for those deploying to ECS to even think about task definitions.
+
+It doesn't matter if your task definitions include environment specific values, or if each environment is represented by a cluster in a new AWS account, as Octopus will create the necessary task definitions on your behalf. This streamlines your workflow to: creating new Docker images, creating an Octopus release with those Docker images, and promoting your release across your environments. It also ensures you can [recover from failed deployments](https://octopus.com/blog/ten-pillars-of-pragmatic-deployments#recoverable-deployments) by redeploying an old release.
```
Comment (adrianhowchin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:94]: This is super cool!

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 12 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:47]: We wanted to call this out early on as it provides a constraint on the first milestone. I'd expect anyone with an interest in our ECS support would be familiar with Fargate, but I'll include a link to the AWS docs for some more context.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 13 above)
Comment (ryangribble) [reply blog/2021-06/rfc-ecs-milestone1/index.md:31]: This is already the case with other "multi hosting targets" such as a Kubernetes cluster... so to me it makes sense to address that issue separately/holistically, while being consistent in modelling similar things in the same way (eg model ECS cluster similar to k8s cluster) so an eventual licensing re-evaluation can address this category/pattern where there are potentially many applications/services being deployed to just one "deployment target".

TLDR: I dont think the addition of ECS support is the vehicle to hitch the "figure out a fairer licensing system" wagon to

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+In summary, the first milestone of our proposed ECS support includes:
+
+* A new target to model environments and tenanted deployments.
+* A new step combining the creation/updating of a task definition and creation/updating of a service.
+* The ability to convert the opinionated step into a raw CloudFormation template.
+* A simplified deployment workflow across environments supporting environment specific variables, channels, and lifecycles.
+
+Thank you for reading this post. We hope you are as excited about the proposed new ECS functionality as we are, and any feedback you may have is greatly appreciated.
+
+Happy deployments!
```
Comment (adrianhowchin) [inline blog/2021-06/rfc-ecs-milestone1/index.md:156]: Is the goal to support ECS, Fargate, or both?

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 13 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:31]: We've circulated the pitch at https://docs.google.com/document/d/1MNgIGoE8Jponw9JknGlZylkNThMwd45IOsdkKf8pOok/edit a few times previously and looped in the sales team for feedback. We haven't recieved any indication that adopting targets is the wrong approach.

An ECS cluster as a target also closly models the approach we've taken with Kubernetes, which has worked reasonably well.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 15 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:156]: Fargate is just one of the technologies backing ECS, with the being EC2 based services. So the distinction here is between EC2 and Fargate deployments, both within the context of ECS. We do plan to support both eventually, but Fragate was the easier choice initially.

Comment (rhysparry) [reply blog/2021-06/rfc-ecs-milestone1/index.md:156]: ECS supports using Fargate to schedule its workloads. It means you can use ECS without having to set up EC2 instances.

Comment (rhysparry) [reply blog/2021-06/rfc-ecs-milestone1/index.md:156]: It also looks like they've added external hosts now too (so hosts of the ECS cluster can exist on-premises). Out of scope for now, but another example of "finding somewhere to run your code".

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (robpearson) [reply blog/2021-06/rfc-ecs-milestone1/index.md]: @mcasperson In the absences of official positioning, I think what you wrote is great. We're planning on working through a positioning process which which make this much easier in the future but it's not done yet. 

Your wording is great in the mean time. 🎉

[the remark this answers is not in the document]
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md]: Cheers, thanks @robpearson

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+author: matthew.casperson@octopus.com
+visibility: public
+published: 2021-06-14-1400
+metaImage: 
+bannerImage: 
+tags:
+ - Product
+---
+
+Octopus is a tool that enables world-class deployments. We've always had industry leading functionality for deploying to your on-premises infrastructure, and we've powered Azure application deployments for years.
```
Comment (TeganAli) [inline blog/2021-06/rfc-ecs-milestone1/index.md:13]: @wordlee do we need the first two paragraphs for background, or do most of our customers know this already?

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+published: 2021-06-14-1400
+metaImage: 
+bannerImage: 
+tags:
+ - Product
+---
+
+Octopus is a tool that enables world-class deployments. We've always had industry leading functionality for deploying to your on-premises infrastructure, and we've powered Azure application deployments for years. 
+
+More recently we expanded into other major cloud service providers, allowing you to use tools like Kubernetes and Terraform to deploy to Azure and AWS. But we can do more — there are still cloud native services that Octopus doesn't provide first class integrations for.
```
Comment (TeganAli) [inline blog/2021-06/rfc-ecs-milestone1/index.md:15]: @wordlee do we need the first two paragraphs for background? Or could they be deleted to shorten the intro?

File: blog/2021-06/rfc-ecs-milestone1/index.md [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+
+The proposed step above is a high level look at where we see this new feature going, but the first milestone will likely have the following limitations:
+
+* Limiting the step to deploying to Fargate only.
+* Only supporting rolling deployments, and not supporting integrated CodeDeploy blue/green deployments.
+* Not providing the ability to build a new load balancer.
+* Exclude auto-scaling settings.
+* Exclude App Mesh and FireLens settings.
+* Exclude service auto-discovery settings.
+* Only creating a service, and not supporting tasks or scheduled tasks.
```
Comment (TeganAli) [inline blog/2021-06/rfc-ecs-milestone1/index.md:152]: @wordlee I may have edited this section incorrectly, can you please check?

**Review by wordlee — APPROVED** [review]
Looks good to me. I added some minor line edits

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 17 above)
Comment (wordlee) [reply blog/2021-06/rfc-ecs-milestone1/index.md:15]: Unless @mcasperson objects, I think we could remove them.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 18 above)
Comment (wordlee) [reply blog/2021-06/rfc-ecs-milestone1/index.md:151]: The edits look fine.

File: blog/2021-06/rfc-ecs-milestone1/index.md (hunk 17 above)
Comment (mcasperson) [reply blog/2021-06/rfc-ecs-milestone1/index.md:15]: These were more for internal teams who may not understand the goal of the steps team. I'd like to keep these in to provide context.

Comment (TeganAli) [reply blog/2021-06/rfc-ecs-milestone1/index.md:15]: No worries, thanks for explaining @mcasperson. Happy to keep those paragraphs in :)
